channeldb/kvdb/etcd/readwrite_cursor_test.go

```diff
@@ -289,5 +289,78 @@ func TestReadWriteCursor(t *testing.T) {
 		vkey("da", "apple"): "3",
 		vkey("f", "apple"):  "5",
 	}
-	assert.Equal(t, expected, f.Dump())
+	require.Equal(t, expected, f.Dump())
+}
+
+func TestReadWriteCursorWithBucketAndValue(t *testing.T) {
```

Comment:
To be fair if we want `bbolt` compatibility then it's better to implement `bbolt` and use their test suite :) It would be great to have just one test suite though, but that'd require us forking over the `walletdb` interface and tests. Would you like me  to add that to this PR too?

Reply:
> perhaps this test (and test suite) should be written at the `kvdb` level rather than the `etcd` level, otherwise how can we be sure it matches bbolt? probably would have helped catch this sort of inconsistency

Agreed. I think that is best to do in a follow up PR though as there's a few more PRs piling up already.

Reply:
done, ptal :)